compass/logon.py

```diff
@@ -52,58 +51,56 @@ def get(self, url, auth_header: bool = False, **kwargs):
             kwargs["headers"] = {**kwargs.get("headers", {}), **headers}
             kwargs["params"] = {**kwargs.get("params", {}), **params}
 
-        return self.session.get(url, **kwargs)
-
-    def post(self, url, **kwargs):
-        Settings.total_requests += 1
-        data = kwargs.pop("data", None)
-        json_ = kwargs.pop("json", None)
-        return self.session.post(url, data=data, json=json_, **kwargs)
+        if session is not None:
+            return session.get(url, **kwargs)
+        else:
+            return super(CompassLogon, self)._get(url, **kwargs)
 
-    def _jk_hash(self):
+    def _jk_hash(self) -> str:
```

Comment:
This looks horrid. Not related to this PR - but why is this passed by the client?

Reply:
Assume you're asking about the entire `_jk_hash()` method?

Way back in 2018 when I was initially playing around this blocked me for an embarrasingly long amount of time, and then I had the bright idea of reading the javascript...

Relevant code is here - we don't replicate exactly (even though utility.py does have the hashCode func) - the comment in the JS indicates it is used as a temporary thing. I have never been able to work out what JK stands for though -- my best guess is `join key`, but no-where in the JS is it written in full (as far as I remember)
https://github.com/the-scouts/compass-interface/blob/master/js/Scouts.js#L1214-L1225

(permalink, 13-01-2020: https://github.com/the-scouts/compass-interface/blob/31b72b1f5940b836848bb1033f92f134c7124e60/js/Scouts.js#L1214-L1225)

Reply:
Yer! I'd love to know what was going through the Compass devs mind when this was developed.

Interested into why this is used on an auth header, and what that actually is used for on the backend. But hey, strange to say the least.